zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -2708,6 +2708,31 @@ public void getEphemerals(AsyncCallback.EphemeralsCallback cb, Object ctx) {
         getEphemerals("/", cb, ctx);
     }
 
+    /**
+     * Synchronous sync. Flushes channel between process and leader.
+     *
+     * @param path the given path
+     * @throws KeeperException If the server signals an error with a non-zero error code
+     * @throws InterruptedException If the server transaction is interrupted.
+     * @throws IllegalArgumentException if an invalid path is specified
+     */
+    public void sync(final String path) throws KeeperException, InterruptedException {
+        final String clientPath = path;
+        PathUtils.validatePath(clientPath);
+
+        final String serverPath = prependChroot(clientPath);
+
+        RequestHeader h = new RequestHeader();
+        h.setType(ZooDefs.OpCode.sync);
+        SyncRequest request = new SyncRequest();
+        SyncResponse response = new SyncResponse();
+        request.setPath(serverPath);
```

Comment:
It would be cleaner to refactor this to share code with the asynchronous version, since everything up to here is identical.

Reply:
I guess it may not worth the effort.

* To cover only above code, we probably have to introduce wrapper struct to cover context for `submitRequest` and `queuePacket`.
* To cover also following code, we probably have to introduce an overload to accept all parameters from asynchronous code and a flag to differentiate origin as `zk.sync("/", null, null")` is valid.

A little duplication probably be a good here.